packages/app-store/routing-forms/pages/forms/[...appPages].tsx

```diff
@@ -32,18 +38,53 @@ import {
 
 import type { inferSSRProps } from "@lib/types/inferSSRProps";
 
-import { FormAction, FormActionsDropdown, FormActionsProvider } from "../../components/FormActions";
-import { getSerializableForm } from "../../lib/getSerializableForm";
+import {
+  createAction,
+  FormAction,
+  FormActionsDropdown,
+  FormActionsProvider,
+} from "../../components/FormActions";
+
+function NewFormButton() {
+  const query = trpc.viewer.eventTypes.getByViewer.useQuery();
+  const { t } = useLocale();
+  const router = useRouter();
+  if (!query.data) return null;
+
+  const profileOptions = query.data.profiles
+    .filter((profile) => !profile.readOnly)
+    .map((profile) => {
+      return {
+        teamId: profile.teamId,
+        label: profile.name || profile.slug,
+        image: profile.image,
+        membershipRole: profile.membershipRole,
+        slug: profile.slug,
+      };
+    });
+
+  return (
+    <CreateButton
+      subtitle={t("create_routing_form_on").toUpperCase()}
+      createFunction={(teamId) => {
+        createAction({ router, teamId: teamId ?? null });
+      }}
+      options={profileOptions}
+    />
+  );
+}
 
 export default function RoutingForms({
-  forms: forms_,
   appUrl,
 }: inferSSRProps<typeof getServerSideProps> & { appUrl: string }) {
   const { t } = useLocale();
   const { hasPaidPlan } = useHasPaidPlan();
+  const router = useRouter();
+
+  const filters = getFiltersFromQuery(router.query);
 
-  const { data: forms, isLoading } = trpc.viewer.appRoutingForms.forms.useQuery(undefined, {
-    initialData: forms_,
```

Comment:
initialData was causing a bug where it was same for all filter queries. It's not needed now as I am doing ssr prefetching.